aurora/core.py

```diff
@@ -1189,73 +1339,341 @@ def check_conservation(self, plot=True, axs=None, plot_resolutions=False):
+            ax1[1, 2].set_ylabel('[#]')
+
+            ax1[2, 0].plot(time_average, reservoirs["parallel_loss"], label="Par. loss to divertor")
+            ax1[2, 0].plot(time_average, reservoirs["divertor_backflow"],
+                           label="Divertor backflow rate")
+            if self.namelist["pump_chamber"]:
+                ax1[2, 0].plot(time_average, reservoirs["pump_leakage"],
+                           label="Pump leakage rate")
+            ax1[2, 0].set_ylabel('[$s^{-1}$]')
+
+            for aa in ax1.flatten()[:7]:
+                aa.legend(loc="best").set_draggable(True)
+
+            for ii in [0, 1, 2]:
+                ax1[2, ii].set_xlabel("Time [s]")
+            ax1[2, 0].set_xlim(time_average[[0, -1]])
+
+            # ----------------------------------------------------------------
+            # plot the impurity radiation, if present:
+            if "impurity_radiation" in reservoirs:
+                if axs is None:
+                    fig, ax2 = plt.subplots()
+                else:
+                    ax2 = axs[1]
+
+                ax2.plot(self.time_out, reservoirs["impurity_radiation"], label="Core radiation")
+                ax2.legend(loc="best").set_draggable(True)
+                ax2.set_xlabel("Time [s]")
+                ax2.set_ylabel('[W]')
+
+            # ----------------------------------------------------------------
+            # now plot all particle reservoirs to check particle conservation:
+            if axs is None:
+                fig, ax3 = plt.subplots()
+            else:
+                ax3 = axs[1]
+
+            ax3.set_xlabel("time [s]")
+
+            ax3.plot(time_average, reservoirs["particles_in_plasma"], label="Particles in the plasma")
+            ax3.plot(time_average, reservoirs["particles_in_divertor"], label="Particles in the divertor chamber")
+            if self.namelist["pump_chamber"]:
+                ax3.plot(time_average, reservoirs["particles_in_pump"], label="Particles in the pump chamber")
+            ax3.plot(time_average, reservoirs["particles_stuck_at_wall"], label="Particle stuck at the main wall")
+            ax3.plot(time_average, reservoirs["particles_retained_at_wall"],
+                     label="Particles retained at the main wall")
+            ax3.plot(time_average, reservoirs["particles_pumped"], label="Particles pumped away")
+            ax3.plot(time_average, reservoirs["total"], label="Total particles in the system")
+            ax3.plot(time_average, reservoirs["integ_source"], label="Integrated source")
         
-            ax2.set_ylim(0, None)
-            ax2.legend(loc="best")
+            ax3.set_xlim(self.time_out[[0, -1]])
+            ax3.set_ylim(0, None)
+            ax3.legend(loc="best")
             plt.tight_layout()
 
         if plot:
-            return out, (ax1, ax2)
+            if "impurity_radiation" in reservoirs:
+                return reservoirs, (ax1, ax2, ax3)
```

Comment:
@antonellozito at the moment it seems that the docstring is not consistent with this change that you made, where you return 3 axes. Also, I would say that having many different axes returned by a function is already questionable practice, and here it's being done even more...

Reply:
@fsciortino this is something I forgot to do. When we decide how to organize the plots (see previous answer), I'll update the docstring consequently.